queue/src/main/java/com/bazaarvoice/emodb/queue/core/kafka/KafkaProducerService.java

```diff
@@ -0,0 +1,66 @@
+package com.bazaarvoice.emodb.queue.core.kafka;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.concurrent.Future;
+
+public class KafkaProducerService {
+    private static final Logger _log = LoggerFactory.getLogger(KafkaProducerService.class);
+    private final KafkaProducer<String, String> producer; // Changed to String
+
+    public KafkaProducerService() {
+        this.producer = new KafkaProducer<>(KafkaConfig.getProducerProps());
+        _log.info("KafkaProducerService initialized with producer properties: {}", KafkaConfig.getProducerProps());
+    }
+
+    /**
+     * Sends each message from the collection to the specified Kafka topic separately.
+     *
+     * @param topic   The Kafka topic.
+     * @param events  The collection of messages to be sent.
+     */
+    public void sendMessages(String topic, Collection<String> events, String queueType) {
+        _log.info("Sending {} messages to topic '{}'", events.size(), topic);
+        for (String event : events) {
+            sendMessage(topic, event,queueType);
+        }
+        _log.info("Finished sending messages to topic '{}'", topic);
+    }
+
+    /**
+     * Sends a single message to the specified Kafka topic.
+     *
+     * @param topic   The Kafka topic.
+     * @param message The message to be sent.
+     */
+    public void sendMessage(String topic, String message, String queueType) {
+        ProducerRecord<String, String> record = new ProducerRecord<>(topic, message);
+        try {
+            Future<RecordMetadata> future = producer.send(record, (metadata, exception) -> {
+                if (exception != null) {
+                    _log.error("Failed to send message to topic '{}'. Error: {}", topic, exception.getMessage());
+                }
+            });
+            // Optionally, you can wait for the send to complete
+            RecordMetadata metadata = future.get(); // Blocking call
```

Comment:
This is a blocking call and will impact performance if the `sendMessages` method is called with many messages. It is preferable to let the internal of the Kafka Client batch the calls based on your LingerMS and BatchSize parameters, which won't be leveraged in this case if you call the blocking `get()` method on the [future object](https://kafka.apache.org/0102/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html#send(org.apache.kafka.clients.producer.ProducerRecord,%20org.apache.kafka.clients.producer.Callback)).
I would keep the `sendMessage` as is (for singular message send scenario) and create a private async version (`sendMessageAsync`) that returns a future that the `sendMessages` can use and check for the `Future.isDone()` across all "Futures" for completion. This way you can truly benefit from [Kafka Producer Batching](https://learn.conduktor.io/kafka/kafka-producer-batching/) capability. 
I would be curious to see the current impact to performance of the `sendBatch` function before and after...